### ISSUE TITLE
Add @mouseenter and @mouseleave for QDrawer

### ISSUE THREAD

**Issue by smajl** [quasarframework/quasar#4639]
**Is your feature request related to a problem? Please describe.**
tl;dr: `q-drawer` fires `mouseover` not only for itself but for all its child elements, which is undesired in 99% cases.

It's problematic to switch `q-drawer` back to mini mode by clicking inside it if it works in "mini to overlay" mode. That's because it only supports `@mouseover` and not much more robust `@mouseenter` event. The problem with `@mousover` is that when you click inside drawer and set programmatically `mini = true` then it takes only slight mouse movement to trigger `@mouseover="mini = false"` on `q-drawer` again and it reopens.

**Describe the solution you'd like**
Support `@mouseenter` and `@mouseleave` on `q-drawer`.

**Describe alternatives you've considered**
One would need to wrap the content of the `q-drawer` inside custom `fit` container and attach event handlers on it. It gets even more complicated if you have e.g. `q-scroll-area` inside the drawer. Yuck...

**Comment by pdanpdan** [comment]
Doesn't `@mouseenter.native` and `@mouseleave.native` work? Can you please make a codepen so I can test?

**Comment by rstoenescu** [comment]
`mouseenter` and `mouseleave` will be available in 1.0.5.

**Comment by smajl** [comment]
Ha, thanks guys, `.native` didn't come to my mind, I will try and glad to hear that the new events will be available in 1.0.5. 👍
